lib/semantics/symbol.cc

```diff
@@ -213,6 +214,11 @@ void Symbol::SetType(const DeclTypeSpec &type) {
       details_);
 }
 
+bool Symbol::IsArray() const {
+  const auto *details{std::get_if<ObjectEntityDetails>(&details_)};
```

Comment:
Functions can return arrays.  Do you want `IsArray()` to be true for them too?  If not, maybe `IsObjectArray()` would be better.

Reply:
Thanks, I'll change it to `IsObjectArray`.